### DISCUSSION TITLE
Does TagStudio even work?

### DISCUSSION THREAD

**Discussion by satorkain** [TagStudioDev/TagStudio#457] [Q&A]
and how does it work? I added a folder with files that I can now see in the TagStudio, but no matter what I enter in the search, it doesn't find anything... I just have a list of thousands of files in front of my eyes... what should I do with this now!?

**Comment by 050011-code** [comment]
Yeah, TagStudio does work.
With TagStudio, you have to tag the files yourself, because the tagging method is entirely unique to it. The readme does tell you how to use the program, if you need help. If you have any specific questions don't be afraid to ask :)

**Reply by satorkain** [reply]
No, I understand that files need to be marked. But damn it, first I need to find out of thousands of files those to which I want to assign a tag, but how to do this if the search doesn’t work?!

**Reply by satorkain** [reply]
![изображение](https://github.com/user-attachments/assets/84de9a08-d41b-4994-8f4a-6889f2a2afbf)
83 pages, and how do I find just one of these files to tag?

**Reply by CyanVoxel** [reply]
Reading the readme will answer these questions: https://github.com/TagStudioDev/TagStudio?tab=readme-ov-file#current-features

**Reply by 050011-code** [reply]
Ah yeah gotcha. So basically the search feature is mainly oriented towards searching for tags meaning to search for files you need to use the prefix ```filename:```

![image](https://github.com/user-attachments/assets/6c3da458-cf4f-4cd0-90f4-52b4b5dc2b5a)

**Reply by CyanVoxel** [reply]
From your screenshot I can also see that you have several files with broken links, meaning that their location was moved since you created the library. To relink them go to Tools -> Fix Unlinked Entries to Search & Relink

**Reply by satorkain** [reply]
> Ah yeah gotcha. So basically the search feature is mainly oriented towards searching for tags meaning to search for files you need to use the prefix `filename:`

Thanks, that's better, but not much. Unfortunately, it only finds by the full file name :( I only have a few files named ‘English for Everyone’, but it just shows all 28 thousands files.

**Reply by satorkain** [reply]
> From your screenshot I can also see that you have several files with broken links, meaning that their location was moved since you created the library. To relink them go to Tools -> Fix Unlinked Entries to Search & Relink

I did so, but nothing changed, all the files remain red

**Reply by CyanVoxel** [reply]
> Thanks, that's better, but not much. Unfortunately it only finds files by exact match :( I only have a few files named ‘English for Everyone’, but it just shows all 28 thousands files.

There's currently known issues with spaces in searches: #112
While this issue mostly applies to tag names, filenames are also affected. Since there's no way to wrap multi-word searches in quotations yet (i.e. `filename: "English for Everyone"`) it'll take in each word ("English", "for", "everyone") and make individual matches for those.

**Reply by CyanVoxel** [reply]
> I did so, but nothing changed, all the files remain red

Does it give a number for how many total unlinked files you have? Yet "Searching & Relinking" didn't reconnect them? Can you verify that these file are actually still somewhere in your library directory (D:\english)?
![image](https://github.com/user-attachments/assets/74c0263c-be85-4a6d-9593-2d8763db01a8)

**Reply by satorkain** [reply]
> There's currently known issues with spaces in searches: #112 While this issue mostly applies to tag names, filenames are also affected. Since there's no way to wrap multi-word searches in quotations yet (i.e. `filename: "English for Everyone"`) it'll take in each word ("English", "for", "everyone") and make individual matches for those.

Yes there is such a thing, but in fact it searches also with spaces, but only if the searched words are not in the middle of the name, but at the beginning. For example, it found the file ‘Dorling Kindersley. English for Everyone Course Book Level 1 Beginner. A Complete Self-Study Programme - (English for Everyone) - 2016’ on request “Dorling Kindersley”.

**Reply by satorkain** [reply]
> Does it give a number for how many total unlinked files you have? Yet "Searching & Relinking" didn't reconnect them? Can you verify that these file are actually still somewhere in your library directory (D:\english)?

it says "Unlinked Entries N\A". I haven't moved any files, I just now opened the folder "d:\english"

**Reply by CyanVoxel** [reply]
> it says "Unlinked Entries N\A". I haven't moved any files, I just now opened the folder "d:\english"

If it says N/A then you need to refresh it first - In fact the "Search & Relink" button is grayed out as you can so because it doesn't let you do anything until you refresh

**Reply by satorkain** [reply]
> > Does it give a number for how many total unlinked files you have? Yet "Searching & Relinking" didn't reconnect them? Can you verify that these file are actually still somewhere in your library directory (D:\english)?
> 
> it says "Unlinked Entries N\A". I haven't moved any files, I just now opened the folder "d:\english"

although now it found only 7 unlinked files

**Reply by CyanVoxel** [reply]
> although now it found only 7 unlinked files

It may be worth it to just delete the .TagStudio folder and start again, if you haven't done anything in this library yet. I'm not sure how these files could be marked as unlinked right off the bat unless they were some sort of temporary files on the drive that no longer exist or something like that

**Reply by satorkain** [reply]
> If it says N/A then you need to refresh it first - In fact the "Search & Relink" button is grayed out as you can so because it doesn't let you do anything until you refresh

I click Refresh All, it finds 7 enties, then I click Search and Relink, then I get unlinked 7, I click search again and it finds 7 entries again and so on and so forth many times.

**Reply by CyanVoxel** [reply]
> I click Refresh All, it finds 7 enties, then I click Search and Relink, then I get unlinked 7, I click search again and it finds 7 entries again and so on and so forth many times.

See what I said about recreating the library

**Reply by satorkain** [reply]
> See what I said about recreating the library

I just recreated, and still all the files are red, and it also shows me 7 unlinked records.

**Reply by CyanVoxel** [reply]
> I just recreated, and still all the files are red, and it also shows me 7 unlinked records.

That's *very* strange... do the files in red still exist on disk?

**Reply by satorkain** [reply]
> That's _very_ strange... do the files in red still exist on disk?

Of course, because I just created a library.

**Reply by satorkain** [reply]
> That's _very_ strange... do the files in red still exist on disk?

just created a new library from a different folder (d:/music), and all the files are also red.

**Reply by CyanVoxel** [reply]
> Of course, because I just created a library.

I'm not sure if you're assuming they're there because you don't think anything could've happened to them, or if you actually went and checked. Either way, is there anything strange about the filenames themselves? Any unusual characters, or at least non-ASCII? If so, would you mind sharing one or more of them so I can try and replicate the issue?

**Reply by satorkain** [reply]
> I'm not sure if you're assuming they're there because...

Those files have been there for a long time and I don't touch them. In the "D:/english" folder there are other folders with Cyrillic names, in the "D:/music" folder there is one folder with hieroglyphics, but for some reason ALL files are red. 

I just created a library from one folder (D:/music/ISIS) where there are several files and where all the names are in Latin, but still all the files are red, but now there are no unlinked entries. 
![изображение](https://github.com/user-attachments/assets/fbdde5dd-ac2e-4fa0-9be1-aba78de012a4)
![изображение](https://github.com/user-attachments/assets/bf13cab0-2c7f-4ac3-bd02-04e74788376c)

**Reply by satorkain** [reply]
> If so, would you mind sharing one or more of them so I can try and replicate the issue?

send an archive of files?

**Reply by satorkain** [reply]
and here is a library with fonts without broken links... hmmm
![изображение](https://github.com/user-attachments/assets/273556e5-2897-408d-9114-8d3699c29c5b)

**Reply by CyanVoxel** [reply]
> send an archive of files?

Sorry, I was just referring to the filenames themselves

**Reply by satorkain** [reply]
> If so, would you mind sharing one or more of them so I can try and replicate the issue?

[My.zip](https://github.com/user-attachments/files/16906534/My.zip)

**Reply by CyanVoxel** [reply]
This may be related to #350, but something just seems very off about it

**Reply by CyanVoxel** [reply]
> > If so, would you mind sharing one or more of them so I can try and replicate the issue?
> 
> [My.zip](https://github.com/user-attachments/files/16906534/My.zip)

I'll take a look at these to see if I have the issue on my end

**Reply by satorkain** [reply]
> This may be related to #350, but something just seems very off about it

I sent you an archive, try working with these files

**Reply by CyanVoxel** [reply]
> [My.zip](https://github.com/user-attachments/files/16906534/My.zip)

These files show up normally on my system. I would ask if you could open up a bug report, but I'm honestly not sure what's causing these unique issues here.
![image](https://github.com/user-attachments/assets/bb8d3507-a3b2-47f6-a9ce-d87b59b32e6d)

**Reply by satorkain** [reply]
> These files show up normally on my system. 

this is very strange, because I created a new folder in the root of the disk and threw these files there and they are still red.
![изображение](https://github.com/user-attachments/assets/4f3d2072-ecf7-4983-bcb1-b1723015ddde)

**Comment by CyanVoxel** [comment]
Issue resolved in #461
